components/script/dom/browsingcontext.rs

```diff
@@ -40,15 +38,15 @@ pub struct BrowsingContext {
     /// Indicates if reflow is required when reloading.
     needs_reflow: Cell<bool>,
 
-    /// Stores this context's session history
-    history: DOMRefCell<Vec<SessionHistoryEntry>>,
-
-    /// The index of the active session history entry
-    active_index: Cell<usize>,
-
     /// Stores the child browsing contexts (ex. iframe browsing context)
     children: DOMRefCell<Vec<JS<BrowsingContext>>>,
 
+    /// The current active document.
+    /// Note that the session history is stored in the constellation,
+    /// in the script thread we just track the current active document.
+    active_document: MutNullableHeap<JS<Document>>,
```

Comment:
Is it possible for a `BrowsingContext` to have previous `Documents` stored in session history? Currently Servo doesn't do this, but I am not sure if that is something that the spec requires. (I am not sure if `BrowsingContext` <-> `Document` is a 1-to-1 relationship)

https://html.spec.whatwg.org/multipage/browsers.html#history